drivers/platform/x86/hp-wmi.c

```diff
@@ -275,9 +287,13 @@ static int hp_wmi_als_state(void)
 static int hp_wmi_dock_state(void)
 {
 	int state = 0;
-	int ret = hp_wmi_perform_query(HPWMI_HARDWARE_QUERY, 0, &state,
-				       sizeof(state), sizeof(state));
+	int ret;
+
+	if (dmi_check_system(not_dockable))
+		return 0;
```

Comment:
yeah, the problem is that the `hp_wmi_dock_state()` has no parameters so I should change the function prototype. Also strangely in this driver there is no a device-specific struct where I can add the flag, so I should add it. So I would end up changing a lot of code for a small penalty :)